Shopify/hydrogen
# Integrate hydrogen-react into the hydrogen monorepo

### WHY are these changes introduced?

This is the majority of the work to integrate hydrogen-react back into the hydrogen repo. Namely:

- Code migration (hypothetically, with completely git history. Thanks @gfscott !) This is why the commit history is so large
- Development workflow is working. There's probably room to optimize this, but saving a component in hydrogen-react will cause a page refresh in the demo-store
- CI processes are integrated. In addition, a new monorepo-root-level npm script has been added (called `ci:checks`) that enables you to quickly run all the ci checks locally; before, you had to know which scripts were run in CI, and then run then individually.
- Updates the issue template to incorporate the idea of multiple packages
- Updates the root README to incorporate the idea of multiple packages (thanks @gfscott !)
- Updates docs and package.json to point to the new repo location

A big thank you to @wizardlyhel @juanpprieto and @blittle for pairing on this throughout various points as well!

Final questions:

- [ ] Should we add the NextJS app?



### HOW to test your changes?

1. Checkout this branch, 
2. clear your node_modules at all levels, 
3. `npm i`, 
4. `npm run dev`, 
5.  edit code in hydrogen-react and hydrogen, ensure it works and that it reloads correctly

Also:
- Run ladle (`npm run dev:story`) in the hydrogen-react package and ensure that it is working as well. (This helps you develop components in isolation, outside of remix)

#### Post-merge steps

- [ ] Validate that publishing works with changesets in this new setup
- [ ] Update the docs system to point to this new repo
- [ ] We'll need to migrate issues, PRs, discussions

#### Checklist
